bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/discovery/HomeConnectDiscoveryService.java

```diff
@@ -0,0 +1,149 @@
+import org.openhab.binding.homeconnect.internal.client.model.HomeAppliance;
+import org.openhab.binding.homeconnect.internal.handler.HomeConnectBridgeHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HomeConnectDiscoveryService} is responsible for discovering new devices.
+ *
+ * @author Jonas Brüstel - Initial contribution
+ */
+@NonNullByDefault
+public class HomeConnectDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int SEARCH_TIME = 20;
+
+    private final HomeConnectBridgeHandler bridgeHandler;
+    private final Logger logger;
+
+    /**
+     * Construct an {@link HomeConnectDiscoveryService} with the given
+     * {@link org.openhab.core.thing.binding.BridgeHandler}.
+     *
+     * @param bridgeHandler bridge handler
+     */
+    public HomeConnectDiscoveryService(HomeConnectBridgeHandler bridgeHandler) {
+        super(DISCOVERABLE_DEVICE_THING_TYPES_UIDS, SEARCH_TIME, true);
+        this.bridgeHandler = bridgeHandler;
+        logger = LoggerFactory.getLogger(HomeConnectDiscoveryService.class);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting device scan.");
+
+        HomeConnectApiClient apiClient = bridgeHandler.getApiClient();
+
+        try {
+            List<HomeAppliance> appliances = apiClient.getHomeAppliances();
+            logger.debug("Scan found {} devices.", appliances.size());
+
+            // add found devices
+            for (HomeAppliance appliance : appliances) {
+                @Nullable
+                ThingTypeUID thingTypeUID = getThingTypeUID(appliance);
+
+                if (thingTypeUID != null && DISCOVERABLE_DEVICE_THING_TYPES_UIDS.contains(thingTypeUID)) {
+                    logger.debug("Found {} ({}).", appliance.getHaId(), appliance.getType().toUpperCase());
+                    bridgeHandler.getThing().getThings().forEach(thing -> thing.getProperties().get(HA_ID));
+
+                    Map<String, Object> properties = new HashMap<>();
+                    properties.put(HA_ID, appliance.getHaId());
+                    String name = appliance.getBrand() + " " + appliance.getName() + " (" + appliance.getHaId() + ")";
+
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder
+                            .create(new ThingUID(BINDING_ID, appliance.getType(),
+                                    bridgeHandler.getThing().getUID().getId(), appliance.getHaId()))
+                            .withThingType(thingTypeUID).withProperties(properties)
+                            .withRepresentationProperty(appliance.getHaId())
```

Comment:
Replace `appliance.getHaId()` by `HA_ID`. What is expected is the parameter name, not its value.